Fix error when using worktrees or submodules

abrenneke opened the pull request:
Pass parent `.git` directory to `simple-git` when the git directory is resolved to `.git/worktrees/etc` or `.git/modules/etc`.

Honestly since worktrees are local only things... I think a unit test would be difficult. It'd probably have to set up a worktree, put code into it, and test betterer there? I just made a test repo and verified the change works. If you have other ideas for tests, let me know.

Fixes #786

phenomnomnominal commented:
I wonder if this should be fixed in`findGitRoot`. It claims to be solved here: https://github.com/banyudu/find-git-root/issues/2

abrenneke commented:
Yeah I saw that, looks like they added support in there for following the path that the `.git` file points to... but is that the "git root" or not? I guess not... This is currently blocking two people so getting this asap one way or another would be really helpful 🙏

abrenneke commented:
Hold up, found problem

abrenneke commented:
Okay current PR is confirmed to work now.

`find-git-root` just does it incorrectly, so I just rewrote their function and removed it as a dep. The correct way to handle worktrees and submodules is to basically just ignore it and assume a directory with a `.git` file or folder is the one you should be using.

And one less dep!

phenomnomnominal reviewed (approved):
Heck yeah, looks good to me!
